app/models/campaign.rb

```diff
@@ -0,0 +1,27 @@
+# This "model" is not backed by the database. Its main purpose is giving one of
+# our domain concepts an actual representation in code.
+class Campaign
+  include Singleton
```

Comment:
**thought:** I've always been under the impression of "don't use Singletons", but this setup is _sweet_ 🎉 . Since `SiteConfig` is, more or less DB driven, I don't know of a better way to get use of this "global state". We can't really use an initializer since that would require an app re-boot. It doesn't seem like re-writing a global constant would be better either. This just might be my new go-to reference for a solid use of Singletons 💯

Reply:
> I've always been under the impression of "don't use Singletons",

1. Know the rules.
2. Know when to disregard them.
3. Profit.

As I hinted at in the PR description, I'm not super tied to the singleton pattern here, but this object is essentially a read-only "view" into a DB table with some behavior attached. I could have just made it a module and accessed everything through `Campaign.method`, but then we'd essentially have an implicit singleton and then I'd rather be explicit about it.

Reply:
> Know the rules.
Know when to disregard them.
Profit.

Can I Tweet this? It's gold! 😂

Reply:
Sure, go ahead @mstruve 😃